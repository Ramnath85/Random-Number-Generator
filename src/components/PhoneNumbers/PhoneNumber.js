import React from 'react';

const GeneratedNumbers = ({ phoneNumbers }) => (
  <div>
    { phoneNumbers.length > 1 ?
      <h3><b>Generated Phone Numbers</b></h3>
      :
      <span><i>No phone numbers at the moment</i></span>
    }
    <div>
      <ul className="generated-phone-numbers">
        {
          phoneNumbers && phoneNumbers.map(number => (
            <li key={number}>{number}</li>
          ))
        }
      </ul>
    </div>
  </div>
);

export default GeneratedNumbers;