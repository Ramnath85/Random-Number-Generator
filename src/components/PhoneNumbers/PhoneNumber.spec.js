import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PhoneNumber from "./PhoneNumber";

configure({ adapter: new Adapter() });

describe('Test GeneratedNumbers component', () => {
  let shallowComponent;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      phoneNumbers: []

    };
    shallowComponent = shallow(<PhoneNumber {...defaultProps} />);
  });

  test('should render the PhoneNumber component', () => {
    expect(shallowComponent).toMatchSnapshot();
    expect(shallowComponent.length).toEqual(1)
  });

  test('should text when there are no phone numbers', () => {
    expect(shallowComponent.find('div').find('span').text()).toContain('No phone numbers at the moment')
  });

  test('should render text when there are phone numbers', () => {
    shallowComponent.setProps({
      phoneNumbers : ['0903428765', '0787665456']
    });
    expect(shallowComponent.find('div').find('h3').text()).toContain('Generated Phone Numbers')
  });

  test('should render phone numbers', () => {
    shallowComponent.setProps({
      phoneNumbers : ['0903428765', '0787665456']
    });
    expect(shallowComponent.find('div').find('ul').text()).toContain('0903428765')
  });
});