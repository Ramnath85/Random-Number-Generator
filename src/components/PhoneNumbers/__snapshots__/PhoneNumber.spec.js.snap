// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GeneratedNumbers component should render the PhoneNumber component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GeneratedNumbers
    phoneNumbers={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          <i>
            No phone numbers at the moment
          </i>
        </span>,
        <div>
          <ul
            className="generated-phone-numbers"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i>
            No phone numbers at the moment
          </i>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No phone numbers at the moment",
          },
          "ref": null,
          "rendered": "No phone numbers at the moment",
          "type": "i",
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="generated-phone-numbers"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "generated-phone-numbers",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            <i>
              No phone numbers at the moment
            </i>
          </span>,
          <div>
            <ul
              className="generated-phone-numbers"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i>
              No phone numbers at the moment
            </i>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No phone numbers at the moment",
            },
            "ref": null,
            "rendered": "No phone numbers at the moment",
            "type": "i",
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="generated-phone-numbers"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "generated-phone-numbers",
            },
            "ref": null,
            "rendered": Array [],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
